aplpy/core.py

```diff
@@ -959,14 +959,33 @@ def show_contour(self, data=None, hdu=0, layer=None, levels=5,
         else:
             frame = wcs_contour
 
+        # new var to convert slices tuple to style astropy uses
+        # [0, 3] vs. ['x', 0, , 0, 'y']
+        if slices:
+            new_slices = [0] * frame.pixel_n_dim
+            new_slices[slices[0]] = 'x'
+            new_slices[slices[1]] = 'y'
+        else:
+            new_slices = None
```

Comment:
I think you can make this work without a change to WCSAxes, by slicing the WCS object before you pass it to `get_transform`. Something like this:

```suggestion
        if slices and frame
            new_slices = [0] * frame.pixel_n_dim
            new_slices[slices[0]] = slice(None)
            new_slices[slices[1]] = slice(None)
            frame = frame[new_slices]
        else:
            new_slices = None
```